avocado/core/job.py

```diff
@@ -564,6 +570,9 @@ def run(self):
             self.log.error('Report bugs visiting %s', _NEW_ISSUE_LINK)
             return exit_codes.AVOCADO_FAIL
         finally:
+            for ext in self.job_pre_post_dispatcher.extensions:
+                if hasattr(ext.obj, 'post'):
+                    ext.obj.post(self)
```

Comment:
OK, this is problematic. If you implement `Pre` only job, but it contains `post` method, it's still executed. So I don't see a point in having the explicit Pre/Post interface, if users could just use one class (which is verified on load) and then implement all "optional" methods.

So if this makes you happy, then sure, I can live with it and I'm not reluctant to it (compare to previous version), but I think it's unnecessary step. We can simply define a plugin with no mandatory interface (they are optional) and that's it. If we eventually come-up with some mandatory interfaces, then we should mark them, but currently both, pre are optional and this only makes sure at least one is implemented (and we can do a simple check for it in the dispatcher, if you want to).